packages/arcgis-rest-request/src/job.ts

```diff
@@ -0,0 +1,249 @@
+    const fn = (arg: any) => {
+      this.emitter.off(eventName, fn);
+      handler(arg);
+    };
+
+    this.emitter.on(
+      eventName,
+      fn
+    );
+
+    (handler as any).__arcgis_job_once_original_function__ = fn;
+  }
+
+  off(eventName: string, handler: (e: any) => void) {
+    if ((handler as any).__arcgis_job_once_original_function__) {
+      this.emitter.off(
+        eventName,
+        (handler as any).__arcgis_job_once_original_function__
+      );
+      return;
+    }
+    this.emitter.off(eventName, handler);
+  }
+
+  async getResults(result: string) {
+    const jobInfo = await this.fromExistingJob();
+
+    if (jobInfo.jobStatus === "esriJobSucceeded") {
+      return request(this.jobUrl + "/" + jobInfo.results[result].paramUrl, {
+        authentication: this.authentication
+      });
+    } else {
+      return new Promise((resolve, reject) => {
+        this.startInternalEventMonitoring();
+
+        this.once(JOB_STATUSES.Cancelled, (jobInfo) => {
+          this.stopInternalEventMonitoring();
+          reject(jobInfo);
+        });
+
+        this.once(JOB_STATUSES.TimedOut, (jobInfo) => {
+          this.stopInternalEventMonitoring();
+          reject(jobInfo);
+        });
+
+        this.once(JOB_STATUSES.Failed, (jobInfo) => {
+          this.stopInternalEventMonitoring();
+          reject(jobInfo);
+        });
+
+        this.once(JOB_STATUSES.Success, (jobInfo) => {
+          request(this.jobUrl + "/" + jobInfo.results[result].paramUrl, {
+            authentication: this.authentication
+          })
+            .then((result) => {
+              this.stopInternalEventMonitoring();
+              resolve(result);
+            })
+            .catch((e) => {
+              this.stopInternalEventMonitoring();
+              reject(e);
+            });
+        });
+      });
+    }
+  }
+
+
+
+  // async getAllResults() {
+  //   const results = {
+  //     "out_unassigned_stops": { paramUrl: 'results/out_unassigned_stops' },
+  //     "out_stops": { paramUrl: 'results/out_stops' },
+  //     "out_routes": { paramUrl: 'results/out_routes' }
+  //   };
+
+  //   for (const key in results) {
+  //     const res = await Promise.all([
+  //       request(this.jobUrl + "/results" + key, {
+  //         authentication: this.authentication
+  //       }).then((result) => {
+  //         return result;
+  //       })
+  //   ])
+  //   return res;
+  //   }
+
+  //   //uses key value pairs for the results obj to get each result property to do their own request
+  //   // Promise.all will do all the individual requests and wait until their all done
+  //   // return a object with all the individual result property
+  // };
+
+  cancelJob() {
+    return request(this.jobUrl + "/cancel", {
+      authentication: this.authentication,
+      params: { jobId: this.jobId, returnMessages: false }
+    }).then((response) => this.emitter.emit("cancelled", response));
```

Comment:
@marcysilverman this will return the results of `this.emitter.emit` to the next promise. This should be
```suggestion
    }).then((response) => {
      this.emitter.emit("cancelled", response));
      return response;
    }
```

The test for this should probably also check that the response is passed through.